.github/workflows/UnitTests.yml

```diff
@@ -31,47 +31,50 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        tpu-type: ["v4-8"]
-    name: "TPU test (${{ matrix.tpu-type }})"
-    runs-on: ["self-hosted", "tpu", "${{ matrix.tpu-type }}"]
+        accelerator-type: ["tpu-v4-8","gpu-a100-40gb-4"]
+    name: "Unit test on (${{ matrix.accelerator-type }})"
+    runs-on: ["self-hosted", "${{ matrix.accelerator-type }}"]
     steps:
     - uses: actions/checkout@v3
     - name: Cleanup old docker images
       run: |
         docker system prune --all --force
-    - name: Install dependencies
+    - name: Install TPU dependencies
+      if: ${{ startsWith(matrix.accelerator-type, 'tpu') }}
       run: |
-        bash docker_build_dependency_image.sh
+        bash docker_build_dependency_image.sh DEVICE=tpu
+    - name: Install GPU dependencies
+      if: ${{ startsWith(matrix.accelerator-type, 'gpu') }}
+      run: |
+        bash docker_build_dependency_image.sh DEVICE=gpu
     - name: Analysing the code with pylint
       run: |
         docker run -v /home/runner/actions-runner/_work/maxtext/maxtext:/app --rm --privileged maxtext_base_image bash -c "pylint MaxText/"
-    - name: Test with pytest
+    - name: Test with pytest on TPU
+      if: ${{ startsWith(matrix.accelerator-type, 'tpu') }}
       run: |
         docker run -v /home/runner/actions-runner/_work/maxtext/maxtext:/app --rm --privileged maxtext_base_image bash -c 'cd MaxText;python3 -m pytest'
+    - name: Test with pytest on GPU
+      if: ${{ startsWith(matrix.accelerator-type, 'gpu') }}
+      run: |
+        docker run -v /home/runner/actions-runner/_work/maxtext/maxtext:/app --rm --privileged maxtext_base_image bash -c 'cd MaxText;python3 -m pytest -m "not tpu"'
     - name: Test train.py
       run: |
-        docker run -v /home/runner/actions-runner/_work/maxtext/maxtext:/app --rm --privileged maxtext_base_image bash -c \
-        'python3 MaxText/train.py MaxText/configs/base.yml run_name=runner_$(date +%Y-%m-%d-%H-%M) base_output_directory=gs://runner-maxtext-logs dataset_path=gs://maxtext-dataset steps=2 enable_checkpointing=false'
+        docker run -v /home/runner/actions-runner/_work/maxtext/maxtext:/app --rm --privileged --shm-size=1g maxtext_base_image bash -c \
```

Comment:
It seems like we need separate things we test on GPU and TPU (and perhaps a shared file we test on both?). We shouldn't be eroding the usefulness of the TPU tests to support GPU's -- we should be testing the full capabilities of both systems..

Reply:
Ok --shm-size=1g seems fine but then make sure you set that in https://github.com/google/maxtext/blob/main/docker_build_dependency_image.sh#L64

Michelle, it is really annoying to no longer test the frontier settings. Can I suggest you refactor so it can run "flash_tpu" on TPUs and "flash_gpu" on GPUs. I imagine we can detect if we have GPUs or TPUs attached....